src/version/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+
+find_package(Git QUIET)
+
+if(Git_FOUND AND EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/../../.git")
+    execute_process(COMMAND ${GIT_EXECUTABLE} fetch --all --tags --force)
+    execute_process(COMMAND ${GIT_EXECUTABLE} describe --exclude latest --abbrev=0
+                    OUTPUT_VARIABLE SD2PSX_VERSION)
+    execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --short HEAD
+                    OUTPUT_VARIABLE SD2PSX_COMMIT)
+    execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --abbrev-ref HEAD
+                    OUTPUT_VARIABLE SD2PSX_BRANCH)
+    string(REGEX REPLACE "\n$" "" SD2PSX_VERSION "${SD2PSX_VERSION}")
+    string(REGEX REPLACE "\n$" "" SD2PSX_COMMIT "${SD2PSX_COMMIT}")
+    string(REGEX REPLACE "\n$" "" SD2PSX_BRANCH "${SD2PSX_BRANCH}")
+else()
+    set(SD2PSX_VERSION "None")
+    set(SD2PSX_COMMIT "None")
+    set(SD2PSX_BRANCH "None")
+endif()
+    
+
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/template/version.c ${CMAKE_CURRENT_SOURCE_DIR}/version.c @ONLY)
```

Comment:
lets not create outputs under the source dir - put it under build dir instead. as an alternative, can keep the C file and have it #include an auto-generated .h file (in case it's difficult to compile things under build directly)

Reply:
Moved to build dir